module/VuFind/src/VuFind/Db/Table/User.php

```diff
@@ -104,6 +104,19 @@ public function createRowForUsername($username)
         return $row;
     }
 
+    /**
+     * Retrieve a user object from the database based on the given field name.
+     *
+     * @param string $fieldName  Field name
+     * @param string $fieldValue Field value
+     *
+     * @return UserRow
+     */
+    public function getByField($fieldName, $fieldValue)
```

Comment:
This change also comes at a somewhat unfortunate time, as I'm currently working on introducing an interface that uses standard getters and setters for all user properties, in order to better abstract away from the underlying database. Some of these changes kind of cut against that. But I suppose, worst case scenario, after the refactoring is done, we could introduce a map of field name to getter method in order to keep this working.

Reply:
Since we only need this for id, username and cat_it now, I changed this so that the getter methods are used.

Reply:
Maybe it would be better to move this logic into the UserService. I don't think there's an advantage to adding a new method to the Table class, since the plan is to eliminate the Table class entirely as soon as we can... and this logic could easily be moved into UserService::getUserByField(). Or maybe it's safer still to just add UserService::getUserByUsername() and UserService::getUserByCatalogId() methods and move this switch statement into the calling code. That way, we don't introduce a general-purpose method that may prove difficult to port cleanly, but I don't think it increases the complexity of this PR significantly.